cluster/manager.go

```diff
@@ -0,0 +1,229 @@
+// Care should be taken that this method does not block, since doing
+// so would block the entire UDP packet receive loop. Additionally, the byte
+// slice may be modified after the call returns, so it should be copied if needed.
+func (c *ClusterManager) NotifyMsg(buf []byte) {
+	// we dont have any need for passing messages between nodes, other then
+	// the NodeMeta sent with alive messages.
+	return
+}
+
+// GetBroadcasts is called when user data messages can be broadcast.
+// It can return a list of buffers to send. Each buffer should assume an
+// overhead as provided with a limit on the total byte size allowed.
+// The total byte size of the resulting data to send must not exceed
+// the limit. Care should be taken that this method does not block,
+// since doing so would block the entire UDP packet receive loop.
+func (c *ClusterManager) GetBroadcasts(overhead, limit int) [][]byte {
+	// we dont have any need for passing messages between nodes, other then
+	// the NodeMeta sent with alive messages.
+	return nil
+}
+
+// LocalState is used for a TCP Push/Pull. This is sent to
+// the remote side in addition to the membership information. Any
+// data can be sent here. See MergeRemoteState as well. The `join`
+// boolean indicates this is for a join instead of a push/pull.
+func (c *ClusterManager) LocalState(join bool) []byte {
+	c.Lock()
+	c.Peers[c.node.Name] = c.node
+	meta, err := json.Marshal(c.Peers)
+	c.Unlock()
+	if err != nil {
+		log.Fatal(4, err.Error())
+	}
+	return meta
+}
+
+func (c *ClusterManager) MergeRemoteState(buf []byte, join bool) {
+	knownPeers := make(map[string]Node)
+	err := json.Unmarshal(buf, &knownPeers)
+	if err != nil {
+		log.Fatal(4, err.Error())
+	}
+	c.Lock()
+	for name, meta := range knownPeers {
+		if existing, ok := c.Peers[name]; ok {
+			if meta.Updated.After(existing.Updated) {
+				log.Info("updated node meta found in state update for %s", meta.Name)
+				c.Peers[name] = meta
+			}
+		} else {
+			log.Info("new node found in state update. %s", meta.Name)
+			c.Peers[name] = meta
+		}
+	}
+	c.Unlock()
+}
+
+// Returns true if the node is a ready to accept requests
+// from users.
+func (c *ClusterManager) IsReady() bool {
+	c.RLock()
+	defer c.RUnlock()
+	return c.node.IsReady()
+}
+
+func (c *ClusterManager) SetReady() {
+	c.Lock()
+	if c.node.State == NodeReady {
+		c.Unlock()
+		return
+	}
+	c.node.State = NodeReady
+	c.node.Updated = time.Now()
+	c.Unlock()
+	c.BroadcastUpdate()
+}
+
+func (c *ClusterManager) SetReadyIn(t time.Duration) {
+	go func() {
+		// wait for warmupPeriod before marking ourselves
+		// as ready.
+		time.Sleep(t)
+		c.SetReady()
+	}()
+}
+
+// Returns true if the node is a set as a primary node that should write data to cassandra.
+func (c *ClusterManager) IsPrimary() bool {
+	c.RLock()
+	defer c.RUnlock()
+	return c.node.Primary
+}
+
+// SetPrimary sets the primary status.
+// Note: since we set the primary metric here, this should only be called on ThisNode !
```

Comment:
`c.node` is always ThisNode right, so this comment can be removed. as it's now impossible to call SetPrimary on a node that is not thisNode.